gradio/image_utils.py

```diff
@@ -51,13 +51,17 @@ def format_image(
         )
 
 
-def save_image(y: np.ndarray | PIL.Image.Image | str | Path, cache_dir: str):
```

Comment:
Right now `format_image` will give precedence for the current format of the image instead of the `format` argument. It should be the other way around.